app/src/features/Preview/Preview.tsx

```diff
@@ -138,33 +150,31 @@ const Preview = (): JSX.Element => {
 
   const [apiExploreCreate] = useApiExploreCreateMutation();
 
-  useEffect(() => {
-    if (mappingId && owner && exploration === undefined) {
-      setExploration(null);
-
-      const explore = async () => {
-        try {
-          const exploration = await apiExploreCreate({
-            explorationRequestRequest: {
-              owner: owner?.name ?? "",
-              table: mapping?.primary_key_table ?? "",
-              resource_id: mappingId,
-            },
-          }).unwrap();
-          setExploration(exploration);
-        } catch (e) {
-          setAlerts([e.error]);
-        }
-      };
-      explore();
-    }
-  }, [
-    apiExploreCreate,
-    exploration,
-    mapping?.primary_key_table,
-    mappingId,
-    owner,
-  ]);
+  const handleError = useCallback(
+    (
+      errors: ApiValidationError<unknown> | undefined,
+      errorType: number | "FETCH_ERROR" | "PARSING_ERROR" | "CUSTOM_ERROR",
+      errorField: string
+    ) => {
+      if (errors) {
+        const errorKey = Object.keys(errors);
+        errorKey.forEach((key) => {
+          enqueueSnackbar(
+            t<string>("catchValidationErrorPrompt", {
+              query: errorField,
+              errorStatus: errorType,
+              errorKey: key,
+              errorText: errors[key as keyof typeof errors],
+            }),
+            {
+              variant: "error",
+            }
+          );
+        });
```

Comment:
I suggest you use `Object.entries` instead of `Object.keys`, because of what you do in line 167 => accessing errors[key]

```suggestion
        const errorEntries = Object.entries(errors);
        errorEntries.forEach(([key, text]) => {
          enqueueSnackbar(
            t<string>("catchValidationErrorPrompt", {
              query: errorField,
              errorStatus: errorType,
              errorKey: key,
              errorText: text,
            }),
            {
              variant: "error",
            }
          );
        });
```